### PR TITLE
Feature/add verbose and reset callback

### PR DESCRIPTION
**Pull request by tiagopog**
# What does this PR do?

Brings two additional features to the lib:

1. **Verbose mode:** optional boolean setting to enable the logging of additional texts when spawning the lib's OTP process and when scheduling the next resetting, which is useful for additional debugging. Default is `false` to keep compatibility with previous behavior;
2. **Reset callback:** optional function to be called every time connections are reset in the pool. It's useful for things like tracking resets on APM tools.

Along with that I also added a few more tests cases to improve the overall coverage of the lib.

### CODE REVIEW

File: test/test_helper.exs [hunk 1]
```diff
@@ -16,6 +16,12 @@ defmodule TestPool do
   def pool_type, do: DBConnection.ConnectionPool
 end
 
+defmodule TestDatabaseObserver do
+  def db_connection_reset_callback(_state) do
+    IO.puts("Callback: track connection reset on APM")
```
Comment (grahac) [inline test/test_helper.exs]: Will this cause an error in credo checking?  Maybe we can either make this a Logger or find another way that does not use IO.puts?

**Review by grahac — COMMENTED** [review]
Overall looks good and happy to add these.  Please just check the comment on IO.puts

File: test/test_helper.exs (hunk 1 above)
Comment (tiagopog) [reply test/test_helper.exs]: Credo is actually ok with it:

```elixir
$ mix credo test/test_helper.exs
Checking 1 source file ...

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.3 seconds (0.1s to load, 0.1s running 52 checks on 1 file)
5 mods/funs, found no issues.
```

Anyway, let me replace it by a simple `:ok` atom.

**Comment by tiagopog** [comment]
By the way, thanks for the work on this lib. It's been deployed to our QA and Production web nodes for a couple of days now and it's been working seamlessly.

**Review by grahac — APPROVED** [review]

**Pull request merged**